snuba/datasets/errors_replacer.py

```diff
@@ -206,93 +213,124 @@ def set_project_needs_final(
     project_id: int,
     state_name: Optional[ReplacerState],
     replacement_type: ReplacementType,
-) -> Optional[bool]:
-    key, type_key = get_project_needs_final_key_and_type_key(project_id, state_name)
+) -> None:
+    key, type_key = build_project_needs_final_key_and_type_key(project_id, state_name)
     p = redis_client.pipeline()
-    p.set(key, True, ex=settings.REPLACER_KEY_TTL)
+    p.set(key, time.time(), ex=settings.REPLACER_KEY_TTL)
     p.set(type_key, replacement_type, ex=settings.REPLACER_KEY_TTL)
-    [final_was_set, _] = p.execute()
-    return bool(final_was_set)
+    p.execute()
+
+
+@dataclass
+class ProjectsQueryFlags:
+    needs_final: bool
+    group_ids_to_exclude: Sequence[int]
+    replacement_types: Set[str]
+    latest_replacement_time: Optional[datetime]
 
 
 def get_projects_query_flags(
     project_ids: Sequence[int], state_name: Optional[ReplacerState]
-) -> Tuple[bool, Sequence[int], Set[str]]:
-    """\
+) -> ProjectsQueryFlags:
+    """
     1. Fetch `needs_final` for each Project
     2. Fetch groups to exclude for each Project
-    3. Trim groups to exclude ZSET for each Project
+    3. Clean up the old replacement types and groups
     4. Fetch replacement types for each Project
+    5. Fetch latest exclude groups replacement type for each Project
 
-    Returns (needs_final, group_ids_to_exclude, replacement_types)
+    Returns (
+        needs_final,
+        group_ids_to_exclude,
+        replacement_types,
+        latest_replacement_time
+    )
```

Comment:
This does not add any information with respect to the return type structure. You do not need to explain that here, and it can be counter productive if we update the type and forget to update the comment.

Remove please.